### PR TITLE
[ML] unify usages of _all and wildcard <*>

### PR DESCRIPTION
**Pull request by benwtrent**
Everywhere we use `_all` should also support `*`. 

This commit corrects the few places were this was not explicitly the case.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/Strings.java [hunk 1]
```diff
@@ -746,6 +746,10 @@ public static boolean isAllOrWildcard(String[] data) {
                data.length == 1 && ("_all".equals(data[0]) || "*".equals(data[0]));
     }
 
+    public static boolean isAllOrWildcard(String data) {
```
Comment (przemekwitek) [inline server/src/main/java/org/elasticsearch/common/Strings.java:749]: Please add a comment and unit test.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetCalendarsAction.java [hunk 2]
```diff
@@ -34,7 +35,7 @@ public TransportGetCalendarsAction(TransportService transportService, ActionFilt
     @Override
     protected void doExecute(Task task, GetCalendarsAction.Request request, ActionListener<GetCalendarsAction.Response> listener) {
         final String calendarId = request.getCalendarId();
-        if (request.getCalendarId() != null && GetCalendarsAction.Request.ALL.equals(request.getCalendarId()) == false) {
+        if (request.getCalendarId() != null && Strings.isAllOrWildcard(request.getCalendarId())== false) {
```
Comment (przemekwitek) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetCalendarsAction.java:38]:
```suggestion
        if (request.getCalendarId() != null && Strings.isAllOrWildcard(request.getCalendarId()) == false) {
```

**Comment by droberts195** [comment]
> Everywhere we use `_all` should also support `*`.

I believe the intention is that by 8.0 everywhere we use `_all` should also support more generic wildcarding, for example `foo*` or `foo*,bar*`.  Maybe this will be extremely hard for us in some places, maybe trivial in other places.  It would be great if you could make a list of all the places in ML that don't currently support generic wildcarding like `foo*,bar*` and open a meta issue with checkboxes for each.  Then we can decide what to do about each case, and even if we don't fix them all at least we'll know what's outstanding.

I am happy to merge this PR with just the nits corrected, as it's a step in the right direction.

**Comment by benwtrent** [comment]
run elasticsearch-ci/2

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**